pydicer/convert/convert_pt.py

```diff
@@ -0,0 +1,311 @@
+
+    position_final_slice = (len(slices) - 1) * slice_spacing + slices[
+        0
+    ].ImagePositionPatient[2]
+    # Test whether some slices are missing
+    # Pang -- due to an error at line 144: TypeError: only size-1 arrays can be converted
+    # to Python scalars
+    if not is_approx_equal(
+            position_final_slice, float(slices[-1].ImagePositionPatient[2])
+        ):
+        if (position_final_slice - axial_positions[-1]) / slice_spacing < 1.5:
+            # If only one slice is missing
+            diff = np.asarray(
+                [
+                    not is_approx_equal(
+                        float(axial_positions[ind])
+                        - float(axial_positions[ind - 1])
+                        - slice_spacing,
+                        0,
+                    )
+                    for ind in range(1, len(axial_positions))
+                ]
+            )
+            ind2interp = int(np.where(diff)[0])
+            new_slice = (
+                np_image[:, :, ind2interp] + np_image[:, :, ind2interp + 1]
+            ) * 0.5
+            new_slice = new_slice[..., np.newaxis]
+            np_image = np.concatenate(
+                (np_image[..., :ind2interp], new_slice, np_image[..., ind2interp:]),
+                axis=2,
+            )
+            warnings.warn(
+                "One slice is missing, we replaced it by linear interpolation"
+            )
+        else:
+            # if more than one slice are missing
+            raise RuntimeError("Multiple slices are missing")
+
+    image_position_patient = [float(k) for k in slices[0].ImagePositionPatient]
+    sitk_image = get_sitk_volume_from_np(
+        np_image, pixel_spacing, image_position_patient
+    )
+
+    sitk_writer.SetFileName(output_filepath)
+    sitk_writer.Execute(sitk_image)
+
+    return np.transpose(np_image, (1, 0, 2)), pixel_spacing, image_position_patient
+
+
+def get_sitk_volume_from_np(np_image, pixel_spacing, image_position_patient):
+    """Function to get sitk volume from np image
+    Args:
+    Returns:
+    """
+    trans = (2, 0, 1)
+    sitk_image = sitk.GetImageFromArray(np.transpose(np_image, trans))
+    sitk_image.SetSpacing(pixel_spacing)
+    sitk_image.SetOrigin(image_position_patient)
+    return sitk_image
+
+
+def is_approx_equal(x, y, tolerance=0.05):
+    """Function to know is_approx_equal
+    Args:
+    Returns:
+    """
+    return abs(x - y) <= tolerance
+
+
+class MissingWeightException(RuntimeError):
+    pass
+
+
+def get_physical_values_pt(slices, patient_weight, dtype=np.float32):
+    """Function to Get physical values from raw PET
+    Args:
+    Returns:
+    """
+    s = slices[0]
+    # units = s.Units
+    if "Units" in s:  # from Shuchao's code (default)
+        units = s.Units
+    else:
+        units = "CNTS"
+
+    if units == "BQML":
+
+        acquisition_datetime = datetime.strptime(
+            s[Tag(0x00080022)].value + s[Tag(0x00080032)].value.split(".")[0],

```

Comment:
Not sure what these hexadecimal Tags are, but we should probably define them as constants at the top of the script or in a constants file rather than magic numbers like this. This goes for all Tag(...) references

Reply:
Yeah these tags refer to specific DICOM tags. We can access those with a user friendly name.

In this case it would be: `s.AcquisitionDate` and `s.AcquisitionTime`. Please use those user-friendly names.